src/scr_flush_file.c

```diff
@@ -40,6 +45,167 @@
 #error "globals.h accessed from tools"
 #endif
 
+/*
+ * Given a path to a prefix directory, the contents of the flush file,
+ * an id, and the path to the flush_file, generate the summary.scr file and
+ * update flush.scr to show that we're no longer flushing.
+ *
+ * Returns 0 on success.  Returns 1 if there's no corresponding dataset_id
+ * in the flush file.
+ */
+int write_summary_file(char* prefix, kvtree* flush_file, int dataset_id,
+  char* flush_file_path)
+{
+
+  /* define metadata directory for dataset */
+  char* summary_file = NULL;
+  int rc;
+  kvtree* flush_key_dataset;
+  kvtree* summary;
+
+  flush_key_dataset = kvtree_get_kv_int(flush_file, SCR_FLUSH_KEY_DATASET, dataset_id);
+  if (flush_key_dataset == NULL) {
+    printf("%s: no flush file entry for dataset %d\n", __func__, dataset_id);
+    return 1; /* No entry for this dataset */
+  }
+
+  summary = kvtree_get(flush_key_dataset, SCR_FLUSH_KEY_DSETDESC);
+
+  asprintf(&summary_file, "%s/.scr/scr.dataset.%d/summary.scr", prefix, dataset_id);
+
+  rc = scr_flush_summary_file(summary, 1, summary_file);
+
+  scr_free(&summary_file);
+
+  /* Update the flush file that we're not longer transferring */
+  scr_flush_file_location_unset_with_path(dataset_id,
+    SCR_FLUSH_KEY_LOCATION_SYNC_FLUSHING, flush_file_path);
+
+  return rc;
+}
+
+/*
+ * Given a path to a state_file, resume and finalize all transfers for all
+ * files in the state_file.
+ */
+int resume_transfer(char* state_file_path)
+{
+  int id;
+  int rc;
+  AXL_Init();
+
+  id = AXL_Create(AXL_XFER_STATE_FILE, "scr", state_file_path);
+  if (id < 0) {
+    scr_err("Error: AXL_Create() = %d\n", id);
+    rc = id;
+    goto end;
+  }
+
+  rc = AXL_Resume(id);
+  if (rc != AXL_SUCCESS) {
+    scr_err("Error: AXL_Resume(%d) = %d\n", id, rc);
+    goto end;
+  }
+
+  rc = AXL_Wait(id);
+  if (rc != AXL_SUCCESS) {
+    scr_err("Error: AXL_Wait(%d), = %d\n", id, rc);
+    goto end;
+  }
+
+  rc = AXL_Free(id);
+  if (rc != AXL_SUCCESS) {
+    scr_err("Error: AXL_Free(%d), = %d\n", id, rc);
+    goto end;
+  }
+
+  rc = AXL_Finalize();
+  if (rc != AXL_SUCCESS) {
+    scr_err("Error: AXL_Finalize(), = %d\n", id, rc);
+  }
+end:
+  return rc;
```

Comment:
We should double check the return value here for cases above.  I'm guessing we at least want to convert from AXL return codes to something else.  Or we could have it return void since we're not using that from where it's called.

Reply:
I'll just make it a `void` and simplify the function.